### PR TITLE
multi: introduce new meta type for json strings

### LINKED ISSUE #783: spec+asset: move to officially designate asset meta type 1 as a JSON blob
The meta field was always intended to allow developers and users to specify custom meta data for assets. So far, we just have the first type, `0` which is an opaque blob of bytes. Rather than try to enumerate each new possilbe meta type, a strong contender for something more flexible is just a JSON blob here. 

By designating asset meta type `1` as a JSON blob, we also give wallets, explorers and other tools a hint on how to properly parse the asset meta data. 

### Steps To Completion 

- [ ] Update the spec and code to specify the new asset meta type. 

- [ ] Update RPCs and CLI as necessary to provide streamlined developer UX.

### PR DESCRIPTION
**Pull request by Roasbeef**
In this PR, we designate meta type `1` as a JSON string meta. We perform upfront validation at the CLI and RPC level, but retain the meta hash only validation at the protocol level (proof validity only cares about hashing, and not the structure)

Fixes #783

### CODE REVIEW

File: cmd/tapcli/assets.go [hunk 1]
```diff
@@ -142,6 +145,30 @@ func parseAssetType(ctx *cli.Context) (taprpc.AssetType, error) {
 	}
 }
 
+func parseMetaType(metaType string,
+	metaBytes []byte) (taprpc.AssetMetaType, error) {
+
+	switch metaType {
+
+	case "opaque":
```
Comment (guggero) [inline cmd/tapcli/assets.go:153]: Doesn't this change disallow people to play around with custom meta types on the CLI?
Maybe in the default case we should try to parse the type as a number and then use that and only fail if it's not a valid number? So people can still experiment with types > 1.

File: itest/asset_meta_test.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+	ctxb := context.Background()
+	for _, tc := range testCases {
+		t.t.Run(tc.asset.Asset.Name, func(tt *testing.T) {
+			_, err := t.tapd.MintAsset(ctxb, tc.asset)
+			if err != nil {
+				if tc.errString == "" {
+					tt.Fatalf("unexpected error: %v", err)
+				}
+				if !strings.Contains(
```
Comment (guggero) [inline itest/asset_meta_test.go:74]: nit: use `require.ErrorContains`?

**Review by guggero — APPROVED** [review]
Nice, LGTM :tada:

File: itest/asset_meta_test.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+// testAssetMeta tests the validation+parsing logic for asset meta data.
+func testAssetMeta(t *harnessTest) {
+	// In this test, we'll attempt to issue the following assets with
+	// distinct meta type. Within each test case, negative failure
+	// scenarios may exist.
+	jsonType := taprpc.AssetMetaType_META_TYPE_JSON
+	testCases := []struct {
```
Comment (jharveyb) [inline itest/asset_meta_test.go:19]: We could test support for unknown meta types here as well.

**Review by jharveyb — COMMENTED** [review]
LGTM! Some small lint failures.

File: cmd/tapcli/assets.go (hunk 1 above)
Comment (Roasbeef) [reply cmd/tapcli/assets.go:153]: Ah sure, modified to retain that ability, but still allow these first class aliases.

File: itest/asset_meta_test.go (hunk 2 above)
Comment (Roasbeef) [reply itest/asset_meta_test.go:74]: Ah didn't know that existed, swapped to use

File: itest/asset_meta_test.go (hunk 3 above)
Comment (Roasbeef) [reply itest/asset_meta_test.go:19]: done!

**Pull request merged**